plutus-contract/src/Language/Plutus/Contract/Wallet.hs

```diff
@@ -40,14 +39,48 @@ import qualified Wallet.API                  as WAPI
 import           Wallet.Emulator             (Wallet)
 import qualified Wallet.Emulator             as E
 
+{- Note [Submitting transactions from Plutus contracts]
+
+To turn an 'LedgerTxConstraints' value into a valid transaction that can be
+submitted to the network, the contract backend needs to
+
+* Balance it.
+  If the total value of 'tcInputs' + the 'tcForge' field is
+  greater than the total value of 'tcOutputs', then one or more public key
+  outputs need to be added. How many and what addresses they are is up
+  to the wallet (probably configurable).
+  If the total balance 'tcInputs' + the 'tcForge' field is less than
+  the total value of 'tcOutputs', then one or more public key inputs need
+  to be added (and potentially some outputs for the change).
+  If the total value spent is less than what's specified in the 'tcValueSpent'
+  field then the missing value needs to be added to both sides, inputs and
+  outputs.
```

Comment:
The note is outdated because the `tcValueSpent` bit is now covered by `Ledger.Constraints.addMissingValueSpent`. So the `UnbalancedTx` already covers the entire amount that needs to be spent, and additional `Value` needs to be added only to cover the fee / gas cost.

Reply:
I've updated the note.